### ISSUE TITLE
hxcpp.h: No such file or directory

### ISSUE THREAD

**Issue by NicoM1** [underscorediscovery/flow#61]
Hey, I'm trying to build for android again (had it working before windows reinstall) I've got past a lot of the issues but I keep getting errors related to: 

`Error:(6, 19) P:/Git-Projects/VirtualPet/bin/android.build/cpp/obj/android-v7/__pch/haxe/hxcpp.h: No such file or directory`

if I go to that folder there is a single hxcpp.h.gch file, which is actually mentioned in the console output earlier: 

`flow / build - running hxcpp for arch armv7 ...
Creating P:/Git-Projects/VirtualPet/bin/android.build/cpp/obj/android-v7/__pch/haxe/hxcpp.h.gch...`

If you have any ideas, or if I should move this to the hxcpp repo, please let me know,
Nico

**Comment by underscorediscovery** [comment]
You're missing a lot of information needed to figure things out. What version of haxe? What version of hxcpp? What version of the android SDK? etc. 

I know that some stuff was broken on hxcpp in the latest RC stuff and I saw mentions of precompiled headers (PCH) so I would guess it's related to the version of hxcpp.

**Comment by NicoM1** [comment]
sorry yeah, I'm on 3.2.1, and you were right that that error was hxcpp related (I was on the latest, reverted to an older one and that error is gone now). however now I'm getting huge indecipherable errors from NDK, gonna try installing the version that openfl downloads by default as I assume thats one that should work (I downloaded the latest ndk from the website). and I'm on the 6.0 android SDK, gonna try changing NDK first before downgrading that though.

If nothing I do helps I'll post my current errors but they're a big mess so I want to see if I can prune them down first.

Thanks,
Nico

**Comment by underscorediscovery** [comment]
The Android studio setup + setup guide would setup the necessary Android versions, usually the latest SDK target is used. For building the native part, via hxcpp, the NDK is used. I was using ndk-r10e, but r12b is out (I haven't tested if they broke anything there). 

Keep in mind if you used android before snow alpha 2.0, it's changed. You run build once which generates an Android studio project. Open that in Android studio, it sets up the necessary things it needs, and you hit build. Build runs flow (via `flow.gradle` file) which asks hxcpp to build the `.so` files. Then it runs it on device if you chose run.

I can't help you figure it out without any errors/info though!

**Comment by NicoM1** [comment]
yeah, I noted the difference in build, sorted out to open it and such, I'm just waiting for an NDK to install, if that fails (or doesn't) I'll give more info.

**Comment by NicoM1** [comment]
ok install failed, I'll check that out in a minute, but seemed to build ok, things I had to do:

- revert hxcpp back a bit (chose 3.2.205, unsure when the breaking changes happen, or if you can just disable the precompiled header thing)
- installed r11c ndk, new one didn't seem to be compatible

I can maybe do more testing for you at some point to sort out whats doing it, but my internets pretty slow so installing lots of big toolchains is quite the pain.

**Comment by underscorediscovery** [comment]
That's not a problem, thanks though. I'm busy dealing with a lot of internal/core design process and the new rendering and that stuff is priority. Once all that is in eventually, it will be easy to reassess the state of each target and tighten up all the workflows against all the latest ndk stuff. 

Unfortunately the newest haxe RC and hxcpp releases have just been disruptive. I'll get around to checking on the details soon though.

**Comment by NicoM1** [comment]
ok cool cool, also there is a minor warning on Android n, but the build runs which is what I care about for now (now to sort out scaling, things are not right)
